threshsign/src/ThresholdSignaturesTypes.cpp

```diff
@@ -21,26 +21,29 @@
 Cryptosystem::Cryptosystem(const std::string& sysType,
                            const std::string& sysSubtype,
                            uint16_t sysNumSigners,
-                           uint16_t sysThreshold) {
+                           uint16_t sysThreshold) :
+                             type(sysType),
+                             subtype(sysSubtype),
+                             numSigners(sysNumSigners),
+                             threshold(sysThreshold),
+                             forceMultisig_((numSigners == threshold)?true:false),
+                             signerID(NID),
+                             publicKey("uninitialized")
+{
   if (!isValidCryptosystemSelection(sysType, sysSubtype, sysNumSigners,
         sysThreshold)) {
     throw InvalidCryptosystemException("Invalid cryptosystem selection:"
       " primary type: " + sysType + ", subtype: " + sysSubtype + ", with "
       + std::to_string(sysNumSigners) + " signers and threshold of "
       + std::to_string(sysThreshold) + ".");
   }
-  type = sysType;
-  subtype = sysSubtype;
-  numSigners = sysNumSigners;
-  threshold = sysThreshold;
-  signerID = NID;
 }
 
 // Helper function to generateNewPseudorandomKeys.
 IThresholdFactory* Cryptosystem::createThresholdFactory() {
-  if (type == MULTISIG_BLS_SCHEME) {
+  if (type == MULTISIG_BLS_SCHEME || forceMultisig_) {
```

Comment:
Again, I have preserved the current behaviour

Reply:
I remain a bit dubious about this behavior, but if you are planning to switch to only using multisig soon, I am somewhat inclined to overlook this for the time being because you will presumably be able to remove the need for this unsolicited re-interpretation of the type once we only support one type.

Reply:
Basically in most cases the intention is to use multisig only, but they say there are still some cases when threshold sig is needed.